src/wms/src/test/java/org/geoserver/wms/map/GetMapKvpRequestReaderTest.java

```diff
@@ -9,20 +9,31 @@
 import static org.hamcrest.CoreMatchers.not;
 import static org.junit.Assert.assertThat;
 
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
```

Comment:
Hmmm... ``com.sun``? This is not a public API... however, it does seem to work even in JDK 11, so I guess it won't be a problem short term.
As a possible alternative, see wiremock with random ports: http://wiremock.org/docs/getting-started/
But I would not get crazy about it, it's just a test after all, probably good as is.